mypy/fastparse.py

```diff
@@ -647,13 +652,25 @@ def visit_Num(self, n: ast35.Num) -> Node:
     # Str(string s) -- need to specify raw, unicode, etc?
     @with_line
     def visit_Str(self, n: ast35.Str) -> Node:
-        return StrExpr(n.s)
+        if self.pyversion[0] >= 3 or self.is_stub:
+            # Hack: assume all string literals in Python 2 stubs are normal
```

Comment:
This comment confuses me by speaking of "all string literals". In common parlance, b'x' is still considered a "string literal", even though in the narrower sense it is a "bytes literal". Having digested the code that funnels to this point I think by the time we get here, these are the possibilities:

- Python 2 module (parsed with ast27, then converted with py2to3):
  Str() is only used for u'x' literals
  Bytes() is used for 'x' and b'x' literals (which mean the same in Python 2)

- Python 3 module or stub, or stub for Python 2 (all parsed with ast35):
  Str() is used for 'x' and u'x' literals (which mean the same in Python 3)
  Bytes() is used for b'x' literals

Mypy uses a different mapping (which I wrote down in my earlier comment about the block below). So indeed the mapping here is correct. I would summarize it as follows:

- Python 2
  u'' -> Str() -> UnicodeExpr()
  b'', '' -> Bytes() -> StrExpr()

- Python 3
  b'' -> Bytes() -> BytesExpr()
  u'', '' -> Str() -> StrExpr()